content/src/view.js

```diff
@@ -1350,7 +1351,47 @@ $(window).on('resize.listing', function() {
   }
 });
 
-function updatePaneLinks(pane) {
+
+var searchCache = "file-search";
+var prevSearch;
+function updateSearchResults(search,pane) {
+  search = search.toLowerCase();
+  if (search == prevSearch){
+    return;
+  }
+    
+  if (!state.pane[pane].allLinks){
+    state.pane[pane].allLinks = state.pane[pane].links;
+  }
+    
+  var results;
+  if (search) {
+    var savedCache=cache.get(searchCache,search);
+    if (savedCache) {
+      results = savedCache;
+    } else {
+      results = [];
+      var list = state.pane[pane].allLinks;
+
+      for (j = 0; j < list.length; j++) {
+        if (list[j].name.toLowerCase().indexOf(search) == 0) {
```

Comment:
When `list.length >= 600` (actually, see `server/load.js` `maxRootDirEntries`, we should re-query the `storage` object each time the prefix changes.  That is because the server will never return more than 600 directory entries - but if you apply a prefix, you can retrieve a different set of 600 other names.

This is useful for the root directory.